### PR TITLE
🌱 Bump cluster-api and govmomi, separate flavorgen

### PR DESCRIPTION
**Pull request by chrischdi**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by chrischdi** [comment]
cc @neolit123 , @lubronzhan : with the govmomi bump, govmomi seems to require the correct dns names or IP addresses to be set at the certificate of vCenter when providing the TLS Thumbprint.

File: packaging/go.mod [hunk 1]
```diff
@@ -0,0 +1,87 @@
+module sigs.k8s.io/cluster-api-provider-vsphere/packaging
+
+go 1.22.7
+
+toolchain go1.22.8
```
Comment (sbueringer) [inline packaging/go.mod:5]: Can we avoid setting the toolchain here?

File: Makefile [hunk 2]
```diff
@@ -181,7 +182,7 @@ IMPORT_BOSS_VER := v0.28.1
 IMPORT_BOSS := $(abspath $(TOOLS_BIN_DIR)/$(IMPORT_BOSS_BIN))
 IMPORT_BOSS_PKG := k8s.io/code-generator/cmd/import-boss
 
-CAPI_HACK_TOOLS_VER := 75c986db9e38190a2313eaf6e5f97d955fa96b65 # Note: this is the commit ID of CAPI 75c986db9e38190a2313eaf6e5f97d955fa96b65 from 2024-10-02
+CAPI_HACK_TOOLS_VER := 5d54746e010854a1316e72733e13577d4d5695ec # Note: this is the commit ID of CAPI 5d54746e010854a1316e72733e13577d4d5695ec from 2024-10-02
```
Comment (sbueringer) [inline Makefile:185]: Date is outdated

File: Makefile [hunk 3]
```diff
@@ -486,6 +489,10 @@ verify-modules: generate-modules  ## Verify go modules are up to date
 		git diff; \
 		echo "go module files are out of date"; exit 1; \
 	fi
+	@if !(git diff --quiet HEAD -- go.sum go.mod $(PACKAGING_DIR)/go.mod $(PACKAGING_DIR)/go.sum); then \
```
Comment (sbueringer) [inline Makefile:492]: we don't have to diff top-level go.mod + go.sum again. let's just add the new ones to l.488

File: Makefile [hunk 4]
```diff
@@ -532,6 +539,10 @@ verify-govulncheck: $(GOVULNCHECK) ## Verify code for vulnerabilities
 	if [ "$$R1" -ne "0" ] || [ "$$R2" -ne "0" ]; then \
 		exit 1; \
 	fi
+	$(GOVULNCHECK) -C "$(PACKAGING_DIR)" ./... && R2=$$? || R2=$$?; \
```
Comment (sbueringer) [inline Makefile:542]: Should this use R3?

We also shouldn't duplicate the if

File: Makefile [hunk 5]
```diff
@@ -532,6 +535,10 @@ verify-govulncheck: $(GOVULNCHECK) ## Verify code for vulnerabilities
 	if [ "$$R1" -ne "0" ] || [ "$$R2" -ne "0" ]; then \
 		exit 1; \
 	fi
```
Comment (sbueringer) [inline Makefile:537]:
```suggestion
```

**Comment by sbueringer** [comment]
Thx!

/lgtm
/approve

/hold
Feel free to merge when you're happy with tests

**Pull request merged**